ext/openssl/openssl.c

```diff
@@ -351,12 +351,39 @@ int php_openssl_get_ssl_stream_data_index(void)
 	return ssl_stream_data_index;
 }
 
-/* {{{ INI Settings */
+static PHP_INI_MH(OnUpdateLibCtx)
+{
+#if PHP_OPENSSL_API_VERSION >= 0x30000
+	const char *name = ZSTR_VAL(new_value);
+
+	if (!strcmp(name, "default")) {
+		OPENSSL_G(ctx).libctx = OPENSSL_G(ctx).default_libctx;
+	} else if (!strcmp(name, "custom")) {
+		OPENSSL_G(ctx).libctx = OPENSSL_G(ctx).custom_libctx;
+	} else {
+		int err_type;
+		if (stage == ZEND_INI_STAGE_RUNTIME) {
+			err_type = E_WARNING;
+		} else {
+			err_type = E_ERROR;
+		}
+
+		/* Do not output error when restoring ini options. */
+		if (stage != ZEND_INI_STAGE_DEACTIVATE) {
+			php_error_docref(NULL, err_type, "OpenSSL libctx \"%s\" cannot be found", name);
+		}
+		return FAILURE;
+	}
```

Comment:
You can just use the zend_string APIs which are clearer, and can also do some pointer comparisons if the strings are interned for any reason.
```suggestion
#if PHP_OPENSSL_API_VERSION >= 0x30000
	if (zend_string_equals(new_value, "default")) {
		OPENSSL_G(ctx).libctx = OPENSSL_G(ctx).default_libctx;
	} else if (zend_string_equals(new_value, "custom")) {
		OPENSSL_G(ctx).libctx = OPENSSL_G(ctx).custom_libctx;
	} else {
		int err_type;
		if (stage == ZEND_INI_STAGE_RUNTIME) {
			err_type = E_WARNING;
		} else {
			err_type = E_ERROR;
		}

		/* Do not output error when restoring ini options. */
		if (stage != ZEND_INI_STAGE_DEACTIVATE) {
			php_error_docref(NULL, err_type, "OpenSSL libctx \"%s\" cannot be found", ZSTR_VAL(new_value));
		}
		return FAILURE;
	}
```

Reply:
Thanks. It should be `zend_string_equals_literal` btw... :)